lightning-transaction-sync/Cargo.toml

```diff
@@ -19,16 +19,24 @@ time = []
 esplora-async = ["async-interface", "esplora-client/async", "esplora-client/tokio", "futures"]
 esplora-async-https = ["esplora-async", "esplora-client/async-https-rustls"]
 esplora-blocking = ["esplora-client/blocking"]
-electrum = ["electrum-client"]
 async-interface = []
 
+# dummy feature to enable the common codepaths for electrum
+_electrum = []
+# the 'default' electrum feature, enabling `rustls` with the `aws-lc-rs` crypto provider
+electrum = ["_electrum", "electrum-client/use-rustls"]
+electrum-rustls = ["electrum"]
+
+# this feature enables `rustls` with the `ring` crypto provider
+electrum-rustls-ring = ["_electrum", "electrum-client/use-rustls-ring"]
+
 [dependencies]
 lightning = { version = "0.2.0", path = "../lightning", default-features = false, features = ["std"] }
 lightning-macros = { version = "0.2", path = "../lightning-macros", default-features = false }
 bitcoin = { version = "0.32.2", default-features = false }
 futures = { version = "0.3", optional = true }
 esplora-client = { version = "0.11", default-features = false, optional = true }
-electrum-client = { version = "0.22.0", optional = true }
+electrum-client = { version = "0.22.0", optional = true, default-features = false, features = ["proxy"] }
```

Comment:
nit: for posterity it may be nice to document in the commit message why we're now enabling the `proxy` feature here